src/components/kalman_component/prepare_sf.py

```diff
@@ -81,9 +82,11 @@ def prepare_sf(config_path, period_number, base_directory, nudge_factor):
             posterior_p = xr.load_dataset(gridded_posterior_path)
 
             # Get posterior emissions multiplied up to current period p, and apply nudging
-            current_posterior_emis = (
-                posterior_p["ScaleFactor"] * sf["ScaleFactor"] * original_emis
-            )
+            # Account for non-optimized soil sink by calling get_posterior_emissions()
+            posterior_scale_ds["ScaleFactor"] = posterior_p["ScaleFactor"] * sf["ScaleFactor"]
+            posterior_emis_ds = get_posterior_emissions(original_emis_ds, posterior_scale_ds)
+            current_posterior_emis = posterior_emis_ds["EmisCH4_Total"].isel(time=0, drop=True)
+            
```

Comment:
One point I am unsure of is whether we should be removing the soil sink in the calculation of the nudged scale factor below (line 108). I think the answer is yes, but would like to check what @megan-he, @msulprizio, and especially @djvaron think.

Reply:
This is a good question! I _think_ what we want to do is:

1. Remove soil sink from current posterior and original prior
2. Do weighted average of current posterior and original prior
3. Apply lambda scaling
4. Compute scale factors to apply in next period (i.e., sf["ScaleFactor"] on line 108)